import_export/instance_loaders.py

```diff
@@ -50,15 +50,20 @@ def __init__(self, *args, **kwargs):
         pk_field_name = self.resource.get_import_id_fields()[0]
         self.pk_field = self.resource.fields[pk_field_name]
 
-        ids = [self.pk_field.clean(row) for row in self.dataset.dict]
-        qs = self.get_queryset().filter(**{
-            "%s__in" % self.pk_field.attribute: ids
-            })
+        # If the pk field is missing, all instances in dataset are new
+        # and cache is empty.
+        self.all_instances = {}
+        if self.dataset.dict and self.pk_field.column_name in self.dataset.dict[0]:
+            ids = [self.pk_field.clean(row) for row in self.dataset.dict]
+            qs = self.get_queryset().filter(**{
+                "%s__in" % self.pk_field.attribute: ids
+                })
 
-        self.all_instances = {
-            self.pk_field.get_value(instance): instance
-            for instance in qs
-        }
+            self.all_instances = {
+                self.pk_field.get_value(instance): instance
+                for instance in qs
+            }
 
     def get_instance(self, row):
-        return self.all_instances.get(self.pk_field.clean(row))
+        if self.all_instances:
+            return self.all_instances.get(self.pk_field.clean(row))
```

Comment:
I think you will need a default condition of `return None` here (and a test for this)

Reply:
are implicit None returns not allowed in this codebase?

Reply:
I don't think there are any rules against it, but explicit return would match the style in `ModelInstanceLoader`

Reply:
done